.github/workflows/build-and-deploy.yaml

```diff
@@ -0,0 +1,117 @@
+name: build and deploy
+
+on:
+  push:
+    branches:
+      - '**'
+
+jobs:
+  get_changed_services:
+    runs-on: ubuntu-latest
+    outputs:
+      matrix: ${{ steps.set-matrix.outputs.matrix }}
+      matrix_empty: ${{ steps.set-matrix.outputs.matrix_empty }}
+    steps:
+      - name: checkout repo
+        uses: actions/checkout@v1
+
+      - name: install go
+        uses: actions/setup-go@v2
+        with:
+          go-version: '^1.15.6'
+
+      - name: install monobuild
+        run: |
+          go get -d github.com/charypar/monobuild
+          go install github.com/charypar/monobuild@latest
+
+      - name: set services matrix
+        id: set-matrix
+        run: |
+          # See https://github.com/charypar/monobuild#change-detection
+          if [[ $GITHUB_REF == refs/heads/master ]]; then
+            # Main branch mode
+            schedule=$(monobuild diff --main-branch)
+          else
+            # Feature branch mode
+            schedule=$(monobuild diff --base-branch origin/master)
+          fi
+          dependencies=$(echo "$schedule" | awk 'BEGIN { FS = ":" } ; { print $1 }')
+          list="["
+          for dep in ${dependencies[@]}; do
+            service=$(basename $dep)
+            list="$list\"$service\",";
```

Comment:
Yeah that's a good idea, I also don't like my implementation of it. Perhaps we could split it out to it's own action?

Reply:
I wasn't intending to be as ambitious as that, but yeah: I guess this whole job could be an action that outputs `matrix`.

I probably wouldn't bother with `matrix_empty`, and instead use `if: ${{ needs.get_changed_services.outputs.matrix != '[]' }}` or something?

(optionally: If you decide to keep `matrix_empty`, I'd suggest flipping its polarity, and calling it `changes_found` or something. In general, I try to avoid variables that have negatives in their names, because reasoning about double-negatives forces me to engage my brain more than I'd like)